src/index.ts

```diff
@@ -147,12 +154,11 @@ async function bundleMDX(
 
           switch (fileType) {
             case 'mdx': {
-              // I do not want to take the time to type mdx...
-              // eslint-disable-next-line
-              const result = (await createCompiler({
-                remarkPlugins,
-              }).process(contents)) as {contents: string}
-              return {contents: result.contents, loader: 'jsx'}
+              const vfile = await compileMDX(
+                {path: filePath, contents},
+                {jsx: false, remarkPlugins},
```

Comment:
It might be good to pass more options through. Especially `rehypePlugins`.
You could also set `providerImportSource` to `'@mdx-js/react'` (or let users do that) to allow context based component “passing” / “cascading”.

Reply:
This is what I was thinking about. I pass components to my `<MDXProvider>` to replace `a` and `p` with something custom.

#10 adds rehype to the current system so I suggested there it should be added to this PR.

Should we add something like `esbuildOptions`? `rehypePlugins` and `remarkPlugins` aswell as `globals` make sense as _front and crentre_ options then we could have `xdmOptions` to completely customize the options passed to `xdm`?

Reply:
Yup, I had a similar thought. This has been added 👍